### PR TITLE
MCGA-02: create table and shared components

### PR DESCRIPTION
**Pull request by matescarabino**
- Create products table with the data fetched from the DB
- Add edit and delete buttons to each table row
- Create the deleteProduct function
- Create an addProduct button
- Link add and edit buttons to the form route
- Create some shared components such as the Modal and the Button

### CODE REVIEW

**Comment by matescarabino** [comment]
El modal no es un componente compartido todavia... pero la idea es cambiarlo por eso esta en la carpeta shared

File: src/Components/Products/index.js [hunk 1]
```diff
@@ -1,10 +1,108 @@
 import React from 'react'
```
Comment (BareaMartin) [inline src/Components/Products/index.js]: This file should be at a folder structure named "Screens"

File: src/Components/Products/index.js [hunk 2]
```diff
@@ -1,10 +1,108 @@
...
 import styles from './products.module.css';
+import Modal from '../Shared/Modal/index';
+
+const Products = (props) => {
+  const [products, setProducts] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [productId, setProductId] = useState(undefined);
+
+  useEffect(() => {
+    async function fetchProducts() {
```
Comment (BareaMartin) [inline src/Components/Products/index.js]: This function should be in a thunk and storing the response in redux not a local state

File: src/Components/Products/index.js [hunk 3]
```diff
@@ -1,10 +1,108 @@
...
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/products/`);
+        const json = await response.json();
+        setProducts(json.data);
+      } catch (error) {
+        alert('Could not GET Products.', error);
+      }}
+    fetchProducts();
+  }, []);
+
+  const deleteProduct = async (id) => {
```
Comment (BareaMartin) [inline src/Components/Products/index.js]: Same concept here, move this to a thunk to delete products

File: src/Components/Products/index.js [hunk 4]
```diff
@@ -1,10 +1,108 @@
...
+  const [productId, setProductId] = useState(undefined);
+
+  useEffect(() => {
+    async function fetchProducts() {
+      try {
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/products/`);
+        const json = await response.json();
+        setProducts(json.data);
+      } catch (error) {
+        alert('Could not GET Products.', error);
```
Comment (BareaMartin) [inline src/Components/Products/index.js]: handling this error with an alert isn't the best practice, remember alerts blocks the directional flow of work in js, blocking tasks, so you might want to just throw the error and use some cleaner way to show it, could be only a string in red

**Review by BareaMartin — CHANGES_REQUESTED** [review]

File: src/Components/Products/index.js (hunk 1 above)
Comment (matescarabino) [reply src/Components/Products/index.js]: Done

File: src/Components/Products/index.js (hunk 2 above)
Comment (matescarabino) [reply src/Components/Products/index.js]: To/do on other ticket

File: src/Components/Products/index.js (hunk 3 above)
Comment (matescarabino) [reply src/Components/Products/index.js]: To/do on other ticket

File: src/Components/Products/index.js (hunk 4 above)
Comment (matescarabino) [reply src/Components/Products/index.js]: This will be an modal later on. It is like this now so we could know if there is an error

**Review by BareaMartin — APPROVED** [review]
LGTM remember to move async data logic to redux-thunks

**Pull request merged**